src/Experiment/src/CustomAttributeBuilder.cs

```diff
@@ -0,0 +1,581 @@
+                //if (namedField.DeclaringType != con.DeclaringType
+                //    && (!(con.DeclaringType is TypeBuilderInstantiation))
+                //    && !con.DeclaringType!.IsSubclassOf(namedField.DeclaringType!))
+                //{
+                //    // Might have failed check because one type is a XXXBuilder
+                //    // and the other is not. Deal with these special cases
+                //    // separately.
+                //    if (!TypeBuilder.IsTypeEqual(namedField.DeclaringType, con.DeclaringType))
+                //    {
+                //        // IsSubclassOf is overloaded to do the right thing if
+                //        // the constructor is a TypeBuilder, but we still need
+                //        // to deal with the case where the field's declaring
+                //        // type is one.
+                //        if (!(namedField.DeclaringType is TypeBuilder) ||
+                //            !con.DeclaringType.IsSubclassOf(((TypeBuilder)namedFields[i].DeclaringType!).BakedRuntimeType))
+                //            throw new ArgumentException(SR.Argument_BadFieldForConstructorBuilder);
+                //    }
+                //}
+
+                // Make sure the field's type can take the given value.
+                // Note that there will be no coercion.
+                if (fieldValue != null)
+                {
+                    VerifyTypeAndPassedObjectType(fldType, fieldValue.GetType(), $"{nameof(fieldValues)}[{i}]");
+                }
+
+                // First a byte indicating that this is a field.
+                writer.Write((byte)CustomAttributeEncoding.Field);
+
+                // Emit the field type, name and value.
+                EmitType(writer, fldType);
+                EmitString(writer, namedField.Name);
+                EmitValue(writer, fldType, fieldValue);
+            }
+
+            // Create the blob array.
+            m_blob = ((MemoryStream)writer.BaseStream).ToArray();
+        }
+
+        // Check that a type is suitable for use in a custom attribute.
+        private bool ValidateType(Type t)
+        {
+            if (t.IsPrimitive)
+            {
+                return t != typeof(IntPtr) && t != typeof(UIntPtr);
+            }
+            if (t == typeof(string) || t == typeof(Type))
+            {
+                return true;
+            }
+            if (t.IsEnum)
+            {
+                switch (Type.GetTypeCode(Enum.GetUnderlyingType(t)))
+                {
+                    case TypeCode.SByte:
+                    case TypeCode.Byte:
+                    case TypeCode.Int16:
+                    case TypeCode.UInt16:
+                    case TypeCode.Int32:
+                    case TypeCode.UInt32:
+                    case TypeCode.Int64:
+                    case TypeCode.UInt64:
+                        return true;
+                    default:
```

Comment:
Is this just the case for nint and nuint?

Reply:
This is straight from Reflection.Emit.